File: app/src/main/java/vds/developer/aceworkout/MainActivity.java
package vds.developer.aceworkout;

import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.PagerAdapter;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import vds.developer.aceworkout.Adapters.BottomNavPagerAdapter;
import vds.developer.aceworkout.fragments.SummaryFragment;

public class MainActivity extends AppCompatActivity {


    private FragmentManager fragmentManager;
    private TabLayout tabLayout;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ViewPager viewPager = findViewById(R.id.main_fragment);
        BottomNavPagerAdapter pagerAdapter = new BottomNavPagerAdapter(getSupportFragmentManager());
        viewPager.setAdapter(pagerAdapter);

        tabLayout = findViewById(R.id.bottom_tab_slider);
        tabLayout.setupWithViewPager(viewPager);
        setUpTab();
    }


    private void setUpTab() {
        for (int index = 0; index<tabLayout.getTabCount(); index ++) {
            switch(index) {
                case 0 :
                    addTab(index, R.string.report, R.drawable.ic_launcher_foreground);
                    break;
                case 1 :
                    addTab(index, R.string.summary,R.drawable.ic_launcher_foreground);
                    break;
                case 2 :
                    addTab(index, R.string.training, R.drawable.ic_launcher_foreground);
                    break;
                case 3 :
                    addTab(index, R.string.setting, R.drawable.ic_launcher_foreground);
                    break;
                default : addTab(index, R.string.summary, R.drawable.ic_launcher_foreground);
            }
        }
        tabLayout.getTabAt(1).select();
    }

    private void addTab(int index, int titleID, int iconID ) {
        tabLayout.getTabAt(index).setText(titleID).setIcon(iconID);
    }


    private void switchFragments(Fragment fragment){
        if (fragmentManager == null) fragmentManager = getSupportFragmentManager();
        fragmentManager.beginTransaction()
                .replace(R.id.main_fragment, fragment)
                .addToBackStack(null)
                .commit();

    }
}

File: app/src/main/java/vds/developer/aceworkout/MainActivity.kt
package vds.developer.aceworkout

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.fragment.app.FragmentManager
import com.google.android.material.tabs.TabLayout
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import vds.developer.aceworkout.activity.trainingday.TrainingFragment
import vds.developer.aceworkout.db.repository.TrainingDayRepository
import java.time.DayOfWeek
import java.time.LocalDate

class MainActivity : AppCompatActivity() {

    private var fragmentManager: FragmentManager? = null
    private var tabLayout: TabLayout? = null
    var isCalendar = true



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        switchFragments();
        topAppBar.setNavigationOnClickListener {
            // Handle navigation icon press
            drawer.openDrawer(GravityCompat.START)
        }
        initalize_week_btn()
    }

    private fun initalize_week_btn() {
        CoroutineScope(Dispatchers.Default).launch {
            var trainingDayRepository = TrainingDayRepository(application)
            var trainingDaysThisWeek = trainingDayRepository.getWeekTrainingDays(LocalDate.now())
            for (trainingDayEntity in trainingDaysThisWeek) {
                when(trainingDayEntity.dateTime.localDate.dayOfWeek) {
                    DayOfWeek.MONDAY -> {
                        btn_monday.setBackgroundColor(Color.GREEN)
                    }
                    DayOfWeek.TUESDAY -> {
                        btn_tuesday.setBackgroundColor(Color.GREEN)
                    }
                    DayOfWeek.WEDNESDAY -> {
                        btn_wednesday.setBackgroundColor(Color.GREEN)
                    }
                    DayOfWeek.THURSDAY -> {
                        btn_thursday.setBackgroundColor(Color.GREEN)
                    }
                    DayOfWeek.FRIDAY -> {
                        btn_friday.setBackgroundColor(Color.GREEN)
                    }
                    DayOfWeek.SATURDAY -> {
                        btn_saturday.setBackgroundColor(Color.GREEN)
                    }
                    DayOfWeek.SUNDAY -> {
                        btn_sunday.setBackgroundColor(Color.GREEN)
                    }
                }
            }
        }

    }


    private fun switchFragments() {
        if (fragmentManager == null) fragmentManager = supportFragmentManager
        fragmentManager!!.beginTransaction()
                .replace(R.id.main_fragment, TrainingFragment())
                .addToBackStack(null)
                .commit()

    }
}
